proposals/0031-hlsl-vector-matrix-operations.md

```diff
@@ -111,43 +111,31 @@ ByteAddressBuffer model;
 vector<float, 3> ApplyNeuralMaterial(vector<half, 8> inputVector) {
   using namespace dx::linalg;
 
-  MatrixRef<
-      DATA_TYPE_E4M3, 32, 8,
-      MATRIX_LAYOUT_INFERENCING_OPTIMAL>
-      matrix0 = {model, 0, 0};
+  MatrixRef<DATA_TYPE_E4M3, 32, 8, MATRIX_LAYOUT_MUL_OPTIMAL> matrix0 = {model,
+                                                                         0, 0};
 
   VectorRef<DATA_TYPE_FLOAT16> biasVector0 = {model, 1024};
 
-  MatrixRef<
-      DATA_TYPE_E4M3,
-      32, 32, MATRIX_LAYOUT_INFERENCING_OPTIMAL>
-      matrix1 = {model, 2048, 0};
+  MatrixRef<DATA_TYPE_E4M3, 32, 32, MATRIX_LAYOUT_MUL_OPTIMAL> matrix1 = {
+      model, 2048, 0};
 
   VectorRef<DATA_TYPE_FLOAT16> biasVector1 = {model, 3072};
 
-  MatrixRef<
-      DATA_TYPE_E4M3,
-      3, 32, MATRIX_LAYOUT_INFERENCING_OPTIMAL>
-      matrix2 = {model, 4096, 0};
+  MatrixRef<DATA_TYPE_E4M3, 3, 32, MATRIX_LAYOUT_MUL_OPTIMAL> matrix2 = {
```

Comment:
Are any of the sizes significant in any way? Or just arbitrary?

Reply:
If this were real code then they'd be super significant to the developer.  For this test they're somewhat arbitrary, although the `3` in this one is intended to ultimately correspond to a rgb color.